p2p/node/api.go

```diff
@@ -188,6 +188,16 @@ func (p *P2PNode) requestAndWait(peerID peer.ID, location common.Location, data
 
 		// Mark this peer as behaving well
 		p.peerManager.MarkResponsivePeer(peerID, location)
+		select {
+		case resultChan <- recvd:
```

Comment:
I don't think this will work. It will take some time to get a response from our peers, but this will skip immediately to the default case if the response isn't ready yet.

Reply:
I think you are misunderstanding the code here. We are not waiting for a response from our peer here - we are sending the received data from the peer onto the resultChan. Instead of waiting for the caller to consume our message from resultChan, we send the data in a non-blocking fashion. The caller/consumer are obligated to consume this data immediately, but even if they don't the channel is buffered so they can consume it at a later time.